tools/devops/automation/templates/release/vs-insertion-prep.yml

```diff
@@ -168,3 +156,37 @@ stages:
             inputs:
               artifactName: ${{ parameters.uploadPrefix }}vsdrop-multitarget-signed
               downloadPath: $(Build.StagingDirectory)\$(MultiTargetVSDropCommitStatusName)
+
+      - script: make -C $(Build.SourcesDirectory)/dotnet version-props
+        displayName: make version props
+
+      - task: DotNetCoreCLI@2
+        displayName: generate and publish BAR manifest
+        inputs:
+          projects: $(Build.SourcesDirectory)/xamarin-macios/dotnet/package/Microsoft.iOS.Ref/package.csproj
+          arguments: >-
+            -t:PushManifestToBuildAssetRegistry
+            -p:BuildAssetRegistryToken=$(MaestroAccessToken)
+            -p:NupkgPath=$(Build.SourcesDirectory)/nugets-blob
+            -bl:$(Build.ArtifactStagingDirectory)/maestro-binlogs/generate-bar-manifest.binlog
+        condition: and(succeeded(), eq('${{ parameters.pushNugetsToMaestro }}', 'true'))
+
+      - powershell: |
+          $versionEndpoint = 'https://maestro.dot.net/api/assets/darc-version?api-version=2019-01-16'
+          $darcVersion = $(Invoke-WebRequest -Uri $versionEndpoint -UseBasicParsing).Content
+          $arcadeServicesSource = 'https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json'
+          & dotnet tool update microsoft.dotnet.darc --version "$darcVersion" --add-source "$arcadeServicesSource" --tool-path $(Agent.ToolsDirectory)\darc -v n
+          & $(Agent.ToolsDirectory)\darc\darc add-build-to-channel --default-channels --id $(BARBuildId) --publishing-infra-version 3 --password $(MaestroAccessToken) --azdev-pat $(publishing-dnceng-devdiv-code-r-build-re)
```

Comment:
@rolfbjarne @mandel-macaque I am trying to simplify things here by removing separate platform specific publishing steps.  I am also having a hard time remembering why we needed a separate post-build pipeline to run these publishing steps, so I am trying to move them back into the "Prepare release" stage of the core pipeline to see if that will work.

I had a question with respect to the [platform-specific publishing that we are currently doing](https://github.com/xamarin/xamarin-macios/blob/42c9f3143d0053252e6a251eab9970442c427c4d/tools/devops/automation/templates/build/build-nugets.yml#L37-L43) -- if the build is configured to only build a subset of platforms (let's say iOS and tvOS), will it only produce NuGet packages for that platform? Or will the `not-signed-package` artifact still contain packages for everything?

Reply:
> I had a question with respect to the [platform-specific publishing that we are currently doing](https://github.com/xamarin/xamarin-macios/blob/42c9f3143d0053252e6a251eab9970442c427c4d/tools/devops/automation/templates/build/build-nugets.yml#L37-L43) -- if the build is configured to only build a subset of platforms (let's say iOS and tvOS), will it only produce NuGet packages for that platform? Or will the `not-signed-package` artifact still contain packages for everything?

We'll only build packages for platforms that are enabled. So if `iOS` is not enabled, then `not-signed-package` won't contain any iOS packages.



> Is there a way to manually queue an official build to try to test some of these changes?

I ran the post-build pipeline manually, maybe that works: https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=9647369&view=results

Reply:
> > Is there a way to manually queue an official build to try to test some of these changes?
> 
> I ran the post-build pipeline manually, maybe that works: [dev.azure.com/devdiv/DevDiv/_build/results?buildId=9647369&view=results](https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=9647369&view=results)

No, it didn't work. Then the next option is to (re)name your branch `release-test/*` and push to origin (not a fork). There's no need to create a PR, CI should pick the branch up automatically (although sometimes I've had to manually trigger a build - in https://devdiv.visualstudio.com/DevDiv/_build?definitionId=14411 - for new branches).

Reply:
I'm planning to remove the post-build pipeline as part of these changes if all goes well to simplify things, let me try to push these changes to a new release-test branch and see what falls out 😄

Reply:
Trying a test run here: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=9648397&view=results

Reply:
I'm going to close this in favor of the other test branch I have been working out of which is now working